gui/pages/pre_check.go

```diff
@@ -0,0 +1,294 @@
+	page.layout, err = gtk.BoxNew(gtk.ORIENTATION_VERTICAL, 0)
+	if err != nil {
+		return nil, err
+	}
+
+	// Create scroller
+	page.scroll, err = gtk.ScrolledWindowNew(nil, nil)
+	if err != nil {
+		return nil, err
+	}
+	page.scroll.SetMarginStart(48)
+	page.scroll.SetMarginEnd(48)
+	page.scroll.SetMarginTop(24)
+	page.scroll.SetMarginBottom(24)
+	page.layout.PackStart(page.scroll, true, true, 0)
+
+	// Create list
+	page.list, err = gtk.ListBoxNew()
+	if err != nil {
+		return nil, err
+	}
+	page.list.SetSelectionMode(gtk.SELECTION_NONE)
+	page.scroll.Add(page.list)
+	st, err := page.list.GetStyleContext()
+	if err != nil {
+		return nil, err
+	}
+	st.AddClass("scroller-main")
+
+	page.info, err = setLabel("", "label-info", 0)
+	if err != nil {
+		return nil, err
+	}
+	page.info.SetMarginStart(24)
+	page.info.SetMarginEnd(24)
+	page.info.SetMaxWidthChars(1) // The value does not matter but its required for LineWrap to work
+	page.info.SetLineWrap(true)
+	page.info.SetSelectable(true)
+	page.layout.PackStart(page.info, false, false, 0)
+
+	// Create progress bar
+	page.pbar, err = gtk.ProgressBarNew()
+	if err != nil {
+		return nil, err
+	}
+	page.pbar.SetHAlign(gtk.ALIGN_FILL)
+	page.pbar.SetMarginStart(24)
+	page.pbar.SetMarginEnd(24)
+	page.pbar.SetMarginBottom(12)
+	page.pbar.SetMarginTop(12)
+	page.layout.PackEnd(page.pbar, false, false, 0)
+
+	return page, nil
+}
+
+// IsRequired is just here for the Page API
+func (page *PreCheckPage) IsRequired() bool {
+	return true
+}
+
+// IsDone is just here for the Page API
+func (page *PreCheckPage) IsDone() bool {
+	return false
+}
+
+// GetID returns the ID for this page
+func (page *PreCheckPage) GetID() int {
+	return PageIDPreCheck
+}
+
+// GetSummary returns the summary for this page
+func (page *PreCheckPage) GetSummary() string {
+	return page.GetTitle()
+}
+
+// GetTitle returns the title for this page
+func (page *PreCheckPage) GetTitle() string {
+	return utils.Locale.Get("Checking Prerequisites")
+}
+
+// GetIcon returns the icon for this page
+func (page *PreCheckPage) GetIcon() string {
+	return "emblem-system-symbolic"
+}
+
+// GetConfiguredValue returns nothing here
+func (page *PreCheckPage) GetConfiguredValue() string {
+	return ""
+}
+
+// GetRootWidget returns the root embeddable widget for this page
+func (page *PreCheckPage) GetRootWidget() gtk.IWidget {
+	return page.layout
+}
+
+// StoreChanges is just here for the Page API
+func (page *PreCheckPage) StoreChanges() {}
+
+// ResetChanges begins the pre-check
+func (page *PreCheckPage) ResetChanges() {
+	msg := utils.Locale.Get("Checking Prerequisites.")
+	msg = msg + " " + utils.Locale.Get("Please wait.")
+	page.info.SetText(msg)
+
+	go func() {
+		progress.Set(page)
+		err := ctrl.PreCheck(page.model)
+		if err != nil {
+			text := strings.Split(err.Error(), "\n")[0]
+			text = text + "\n" + utils.Locale.Get("See %s for details.", page.controller.GetOptions().LogFile)
+			text = text + " " + utils.Locale.Get("Exit.")
```

Comment:
I would drop this "Exit" as part of the string since the "Exit" button is the only option available to them.

Reply:
I knew you would say this. I was just trying my luck again :)